hist/hist/src/TH1.cxx

```diff
@@ -8230,11 +8247,17 @@ Double_t TH1::KolmogorovTest(const TH1 *h2, Option_t *option) const
 
       // make nEXPT experiments (this should be a parameter)
       prb3 = 0;
+      if (afunc1) Copy(*h1Expt);
+      if (afunc2) h2->Copy(*h2Expt);
       for (Int_t i=0; i < nEXPT; i++) {
-         h1Expt->Reset();
-         h2Expt->Reset();
-         h1Expt->FillRandom(h1_cpy, (Int_t)esum1);
-         h2Expt->FillRandom(h1_cpy, (Int_t)esum2);
+         if (!afunc1) {
```

Comment:
Can we introduce a comment here, explaining what happens in the loop over over toys and why we don't need to fill the histograms if `afunc1` or `afunc2` respectively? And Maybe it's also good to note that they can't be both `false`, which is checked at the beginning of the function. Otherwise one might wonder why the loop is even implemented also for this case.